test/utilities.go

```diff
@@ -71,27 +335,22 @@ func EqualJson(j1, j2 string) (ans bool, err error) {
 			}
 			// Any other ommision will not be considered as equal
 			err = fmt.Errorf(str)
-			return false, err
+			return err
 		}
 		// If the test has properties not found in the baseline, then they are not equal
 		if strings.Contains(line, "___ADDED___") {
 			err = fmt.Errorf(str)
-			return false, err
+			return err
 		}
 		// If the properties are different, they they are not equal
 		if strings.Contains(line, "___DIFFER___") {
-			err = fmt.Errorf(str)
-			return false, err
+			matches := differRegex.FindStringSubmatch(line)
+			if ! binaryOrStringEqual(matches[2], matches[3]) {
+				return fmt.Errorf(str)
+			}
 		}
 	}
-	// Extra layer of check. This is not expected to be true.
-	// If only difference is removed with false value property, then the difference
-	// will be qualified as SupersetMatch. If not, should be be considered equal.
```

Comment:
The binary/string comparison is incompatible with the superset match, so I needed to remove it.